scripts/github_actions/build_ci_dependencies.bash

```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
+# Build all custom dependencies needed by swupd:
+# - Curl
+# - Bsdiff
+# - libarchive
+#
+# As process is slow, it can be cached. If directory dependencies exists,
+# just install already build dependencies.
+
+set -e
+CORES=2
+
+# If alread exists, reuse
+if [ -d dependencies ]; then
+	pushd dependencies
+	for i in *; do
+		if [ -d "$i" ]; then
+			pushd "$i"
+			sudo make install
+			popd
+		fi
+	done
+
+	popd
+	exit
+fi
+
+mkdir dependencies
+pushd dependencies
+
+# build bsdiff
+wget https://github.com/clearlinux/bsdiff/releases/download/v1.0.2/bsdiff-1.0.2.tar.xz
+tar -xvf bsdiff-1.0.2.tar.xz
+pushd bsdiff-1.0.2 && ./configure --prefix=/usr --disable-tests && make -j$CORES && sudo make install && popd
```

Comment:
missing "" around j$CORES, are you running shellcheck against this script too?

Reply:
shellcheck doesn't  complain. I think it knows the value is hardcoded